src/js/ripple/remote.js

```diff
@@ -148,155 +97,133 @@ function Remote(opts, trace) {
+  }
+  if (!Array.isArray(opts.servers)) {
+    throw new TypeError('servers must be an array');
   }
 
-  (opts.servers || []).forEach(function(server) {
-    self.addServer(server);
-  });
-
-  // This is used to remove Node EventEmitter warnings
-  var maxListeners = opts.maxListeners || opts.max_listeners || 0;
+  this.setMaxListeners(this.max_listeners);
 
-  this._servers.concat(this).forEach(function(emitter) {
-    if (emitter instanceof EventEmitter) {
-      emitter.setMaxListeners(maxListeners);
-    }
+  this.servers.forEach(function(server) {
+    var connection = self.addServer(server);
+    connection.setMaxListeners(self.max_listeners);
   });
 
-  function listenerAdded(type, listener) {
-    if (type === 'transaction_all') {
-      if (!self._transaction_subs && self.isConnected()) {
-        self.request_subscribe('transactions').request();
-      }
-      self._transaction_subs += 1;
-    }
-  };
-
-  this.on('newListener', listenerAdded);
-
-  function listenerRemoved(type, listener) {
-    if (type === 'transaction_all') {
-      self._transaction_subs -= 1;
-      if (!self._transaction_subs && self.isConnected()) {
-        self.request_unsubscribe('transactions').request();
+  function listenersModified(action, event) {
+    // Automatically subscribe and unsubscribe to orderbook
+    // on the basis of existing event listeners
+    if (~Remote.TRANSACTION_EVENTS.indexOf(event)) {
+      switch (action) {
+        case 'add':
+          if (++self._transaction_listeners === 1) {
+            self.requestSubscribe('transactions').request();
+          }
+          break;
+        case 'remove':
+          if (--self._transaction_listeners === 0) {
+            self.requestUnsubscribe('transactions').request();
+          }
+          break;
       }
     }
   };
 
-  this.on('removeListener', listenerRemoved);
-
-  if (opts.storage) {
-    this.storage = opts.storage;
-    this.once('connect', this.getPendingTransactions.bind(this));
-  }
-
-  function pingServers() {
-    self._pingInterval = setInterval(function() {
-      var pingRequest = self.requestPing();
-      pingRequest.on('error', function(){});
-      pingRequest.broadcast();
-    }, opts.ping * 1000);
-  };
-
-  if (opts.ping) {
-    this.on('connect', pingServers);
-  }
-
-  function reconnect() {
-    self.reconnect();
-  };
+  this.on('newListener', function(event) {
+    listenersModified('add', event);
+  });
 
-  //if we are using a browser, reconnect
-  //the servers whenever the network comes online
-  if (typeof window !== 'undefined') {
-    if (window.addEventListener) {
-      // W3C DOM
-      window.addEventListener('online', reconnect);
-    } else if (window.attachEvent) {
-      // IE DOM
-      window.attachEvent('ononline', reconnect);
-    }
-  }
+  this.on('removeListener', function(event) {
+    listenersModified('remove', event);
+  });
 };
 
 util.inherits(Remote, EventEmitter);
 
+Remote.DEFAULTS = {
```

Comment:
> Therefore it does cost us something, which is either the robustness of the library or the ability to refactor it.

This may cost someone else something, but it's unlikely to cost anyone anything. People are free to misuse anything. That said, I have no intention of removing it [Remote.DEFAULTS] any time soon and don't see why we would feel compelled to. If we did, we follow proper procedure, make sure it's in the release notes, etc. etc.. It's not exactly the unavoidable one-way street you're imagining.